components/dropdown/dropdown-content-mixin.js

```diff
@@ -787,9 +789,9 @@ export const DropdownContentMixin = superclass => class extends LocalizeCoreElem
 			}
 		}
 
-		const specialMobileStyle = this.mediaQueryList.matches && this.mobileTray;
-		const mobileTrayRightLeft = this.mediaQueryList.matches && (this.mobileTray === 'right' || this.mobileTray === 'left');
-		const mobileTrayBottom = this.mediaQueryList.matches && (this.mobileTray === 'bottom');
+		const specialMobileStyle = this._useMobileStyling && this.mobileTray;
```

Comment:
There seemed to be some sort of race condition where the _handleMobileResize function detects the screen change and sets the _useMobileStyling attribute to true, but it never gets reflected into an attribute on the dropdown. I've fixed it by changing all the mediaQueryList.matches to checking the property directly, to prevent any of the mismatches that caused the styling issues.